TEP003_commit_message_guidelines.rst

```diff
@@ -0,0 +1,136 @@
+TEP003: Git Policy
+========================
+
+Status
+======
+
+ **Discussion**
+
+Responsible
+===========
+
+@yeganer, @orbitfold
+
+Description
+===========
+
+Commit Message Guidelines
+-------------------------
+
+When to commit? Ask yourself the following questions:
+
+* Do the changes you have implemented represent a single cohesive feature? Break a large task down in to several subtasks.
+* Is the system in a build-able state after the commit? Don't do commits that you know will leave the system in a broken state.
+* Can your commit be safely rolled-back? Will the system be left in a broken state if you roll-back one commit? If so then maybe several commits should be merged.
+
+Preferably a commit should affect one file and do one thing only. For example resist the temptation to fix indentation in an unrelated file or in an unrelated class/method.
```

Comment:
I'd mention `git add -p` since it's very handy in that situation. Something like this:

> If the changes to a file are related to several commits use `git add -p` to selectively stage only the changes you want to commit.